tuxemon/technique/techeffect.py

```diff
@@ -73,44 +61,17 @@ class TechEffect(Generic[ParameterClass]):
     (Monster, "monster_slug")   => a Monster instance will be created
     """
 
-    name: ClassVar[str] = "GenericEffect"
-    param_class: ClassVar[Type[ParameterClass]]
-
-    def __init__(
-        self,
-        move: Technique,
-        user: Monster,
-        parameters: Sequence[Any],
-    ) -> None:
-
-        self.user = user
-        self.move = move
-
-        # if you need the parameters before they are processed, use this
-        self.raw_parameters = parameters
-
-        # parse parameters
-        try:
-            if self.param_class._fields:
-
-                # cast the parameters to the correct type, as defined in cls.valid_parameters
-                self.parameters = cast_parameters_to_namedtuple(
-                    parameters,
-                    self.param_class,
-                )
-            else:
-                self.parameters = parameters
-
-        except ValueError:
-            logger.error(f"error while parsing for {self.name}")
-            logger.error(f"cannot parse parameters: {parameters}")
-            logger.error(f"{self.param_class}")
-            logger.error(
-                "please check the parameters and verify they are correct"
-            )
-            self.parameters = None
-
-        self._done = False
+    name: ClassVar[str]
+    session: Session = field(init=False, repr=False)
+    _done: bool = field(default=False, init=False)
+
+    def __post_init__(self):
+        self.session = local_session
+        for tech in local_session.player.monsters:
+            self.user = tech
+            for moves in tech.moves:
```

Comment:
What's going on in this for loop?

Reply:
I don't understand what it's trying to do

Reply:
before there was Technique (self.move) and Monster (self.user), so yesterday I tried to come up with something (wrong).

Reply:
I think it's solved. I need to test some things.